pylearn2/datasets/svhn.py

```diff
@@ -214,11 +233,11 @@ def split_train_valid(path, num_valid_train=400,
             train_index = list(train_index)
 
             train_x = numpy.concatenate((train_x,
-                                         data['X'][:, :, :, train_index]),
+                                        data['X'][:, :, :, train_index]),
                                         axis=3)
             train_y = numpy.concatenate((train_y, data['y'][train_index, :]))
             valid_x = numpy.concatenate((valid_x,
-                                         data['X'][:, :, :, valid_index]),
+                                        data['X'][:, :, :, valid_index]),
```

Comment:
Are you storing as one-hot instead of int? How does that interact with `IndexSpace` stuff (CC/@bartvm)